lib/temporal/workflow/dispatcher.rb

```diff
@@ -6,21 +6,33 @@ class Dispatcher
 
       def initialize
         @handlers = Hash.new { |hash, key| hash[key] = [] }
+        @query_handlers = Hash.new { |hash, key| hash[key] = {} }
       end
 
       def register_handler(target, event_name, &handler)
         handlers[target] << [event_name, handler]
       end
 
+      def register_query_handler(target, query, &handler)
+        query_handlers[target][query] = handler
+      end
+
       def dispatch(target, event_name, args = nil)
         handlers_for(target, event_name).each do |handler|
           handler.call(*args)
         end
       end
 
+      def process_query(target, query, args)
+        if query == '__cadence_web_list'
```

Comment:
This is also slightly misplaced making the `Dispatcher` even more aware of the high-level logic. This part should probably be handled by the `Workflow::Context`

Reply:
Hmmm, maybe expose a new method on this class to return the target keys.

```
def handler_keys(target)
  handlers[target].keys
end
```
This can be called in executor.rb to add `%w[__stack_trace]` to the list for the special query handle.

Reply:
@antstorm I'm not seeing an obvious home for reaching through to the `Dispatcher` from the `TaskProcessor` to invoke the query handler, aside from potentially delegating `dispatch` through the executor. Thoughts?

Reply:
@dwillett why not track these in a separate array in the `Workflow::Context` which you populate every time a query handler is register?